Orange/distance/_distance.pyx

```diff
@@ -157,40 +157,41 @@ def manhattan_rows(np.ndarray[np.float64_t, ndim=2] x1,
             == len(dist_missing) == len(dist_missing2)
 
     distances = np.zeros((n_rows1, n_rows2), dtype=float)
-    for row1 in range(n_rows1):
-        for row2 in range(n_rows2 if two_tables else row1):
-            d = 0
-            for col in range(n_cols):
-                if mads[col] == -2:
-                    continue
+    with nogil:
+        for row1 in range(n_rows1):
+            for row2 in range(n_rows2 if two_tables else row1):
+                d = 0
+                for col in range(n_cols):
+                    if mads[col] == -2:
+                        continue
 
-                val1, val2 = x1[row1, col], x2[row2, col]
-                if npy_isnan(val1) and npy_isnan(val2):
-                    d += dist_missing2[col]
-                elif mads[col] == -1:
-                    ival1, ival2 = int(val1), int(val2)
-                    if npy_isnan(val1):
-                        d += dist_missing[col, ival2]
-                    elif npy_isnan(val2):
-                        d += dist_missing[col, ival1]
-                    elif ival1 != ival2:
-                        d += 1
-                elif normalize:
-                    if npy_isnan(val1):
-                        d += fabs(val2 - medians[col]) / mads[col] / 2 + 0.5
-                    elif npy_isnan(val2):
-                        d += fabs(val1 - medians[col]) / mads[col] / 2 + 0.5
-                    else:
-                        d += fabs(val1 - val2) / mads[col] / 2
-                else:
-                    if npy_isnan(val1):
-                        d += fabs(val2 - medians[col]) + mads[col]
-                    elif npy_isnan(val2):
-                        d += fabs(val1 - medians[col]) + mads[col]
+                    val1, val2 = x1[row1, col], x2[row2, col]
+                    if npy_isnan(val1) and npy_isnan(val2):
+                        d += dist_missing2[col]
+                    elif mads[col] == -1:
+                        ival1, ival2 = int(val1), int(val2)
+                        if npy_isnan(val1):
+                            d += dist_missing[col, ival2]
+                        elif npy_isnan(val2):
+                            d += dist_missing[col, ival1]
+                        elif ival1 != ival2:
+                            d += 1
+                    elif normalize:
```

Comment:
I considered it, but I can't without replicating the case above, which handles discrete data. Getting rid of one `if` out of many, at a cost of basically duplicating the function is not worth it.

But now I see I could do that in `euclidean_cols` and `manhattan_cols`. Not sure I will, though; it's like splitting the function into two.